src/storage/btree_load.c

```diff
@@ -2211,624 +2226,781 @@ btree_first_oid (THREAD_ENTRY * thread_p, DB_VALUE * this_key, OID * class_oid,
-	{			/* unique index */
-	  /* extract class OID */
-	  ret = or_get_oid (&buf, &this_class_oid);
-	  if (ret != NO_ERROR)
-	    {
-	      goto error;
-	    }
-	}
+  if (or_put_oid (&buf, &sort_args->cur_oid) != NO_ERROR)
+    {
+      return ER_FAILED;
+    }
 
-      /* Get OID */
-      ret = or_get_oid (&buf, &this_oid);
-      if (ret != NO_ERROR)
+  if (BTREE_IS_UNIQUE (sort_args->unique_pk))
+    {
+      if (or_put_oid (&buf, &sort_args->class_ids[cur_class]) != NO_ERROR)
 	{
-	  goto error;
+	  return ER_FAILED;
 	}
+    }
 
-      /* Create MVCC header */
-      BTREE_INIT_MVCC_HEADER (&mvcc_header);
-
-      ret = or_get_mvccid (&buf, &MVCC_GET_INSID (&mvcc_header));
-      if (ret != NO_ERROR)
+  /* Pack insert and delete MVCCID's */
+  if (MVCC_IS_HEADER_INSID_NOT_ALL_VISIBLE (mvcc_header))
+    {
+      if (or_put_mvccid (&buf, MVCC_GET_INSID (mvcc_header)) != NO_ERROR)
 	{
-	  goto error;
+	  return ER_FAILED;
 	}
-
-      ret = or_get_mvccid (&buf, &MVCC_GET_DELID (&mvcc_header));
-      if (ret != NO_ERROR)
+    }
+  else
+    {
+      if (or_put_mvccid (&buf, MVCCID_ALL_VISIBLE) != NO_ERROR)
 	{
-	  goto error;
+	  return ER_FAILED;
 	}
+    }
 
-#if defined(SERVER_MODE)
-      if (MVCC_GET_INSID (&mvcc_header) != MVCCID_ALL_VISIBLE)
+  if (MVCC_IS_HEADER_DELID_VALID (mvcc_header))
+    {
+      if (or_put_mvccid (&buf, MVCC_GET_DELID (mvcc_header)) != NO_ERROR)
 	{
-	  /* Set valid insert MVCCID flag */
-	  MVCC_SET_FLAG_BITS (&mvcc_header, OR_MVCC_FLAG_VALID_INSID);
+	  return ER_FAILED;
 	}
-#else
-      /* all inserted OIDs are created as visible in stand-alone */
-      MVCC_SET_INSID (&mvcc_header, MVCCID_ALL_VISIBLE);
-#endif /* SERVER_MODE */
+    }
+  else
+    {
+      if (or_put_mvccid (&buf, MVCCID_NULL) != NO_ERROR)
+	{
+	  return ER_FAILED;
+	}
+    }
+
+  if (is_btree_ops_log)
+    {
+      _er_log_debug (ARG_FILE_LINE,
+		     "DEBUG_BTREE: load sort found oid(%d, %d, %d)"
+		     ", class_oid(%d, %d, %d), btid(%d, (%d, %d), mvcc_info=%llu | %llu.",
+		     sort_args->cur_oid.volid, sort_args->cur_oid.pageid, sort_args->cur_oid.slotid,
+		     sort_args->class_ids[sort_args->cur_class].volid,
+		     sort_args->class_ids[sort_args->cur_class].pageid,
+		     sort_args->class_ids[sort_args->cur_class].slotid,
+		     sort_args->btid->sys_btid->root_pageid,
+		     sort_args->btid->sys_btid->vfid.volid, sort_args->btid->sys_btid->vfid.fileid,
+		     MVCC_IS_FLAG_SET (mvcc_header, OR_MVCC_FLAG_VALID_INSID)
+		     ? MVCC_GET_INSID (mvcc_header) : MVCCID_ALL_VISIBLE,
+		     MVCC_IS_FLAG_SET (mvcc_header, OR_MVCC_FLAG_VALID_DELID)
+		     ? MVCC_GET_DELID (mvcc_header) : MVCCID_NULL);
+    }
+
+  assert (buf.ptr == PTR_ALIGN (buf.ptr, INT_ALIGNMENT));
+
+  if (sort_args->key_type->type->data_writeval (&buf, dbvalue_ptr) != NO_ERROR)
+    {
+      return ER_FAILED;
+    }
+
+  recdes->length = CAST_STRLEN (buf.ptr - buf.buffer);
+  return NO_ERROR;
+}
+
+static int
```

Comment:
please put a comment as function header to explain the function.

Reply:
I'll add a comment to that function.